src/components/Settings/AddPreferences.tsx

```diff
@@ -1,54 +1,119 @@
 import Locale from "@/locales";
-import { InputField } from "./InputField";
+import { api } from "@/utils/api";
 import { CheckMark } from "./CheckMark";
-import { useState, Dispatch } from "react";
-import { Preferences } from "../../shared/Classes";
+import { InputField } from "./InputField";
+import { Preferences } from "@/shared/Classes";
+import { Dispatch, useState, useEffect } from "react";
+
+type Options = "Vegan" | "Meat" | "LactoseFree" | "GlutenFree";
 
 export function AddPreferences({
   t,
+  pos,
+  type,
   preferences,
   setPreferences,
   editState,
   setEditState,
-  sectionOne,
 }: {
   t: Locale;
-  preferences: Preferences;
-  setPreferences: Dispatch<Preferences>;
-  editState: boolean;
-  setEditState: Dispatch<boolean>;
-  sectionOne: boolean;
+  pos: number;
+  type: "Representative" | "Banquet";
+  preferences: Preferences[];
+  setPreferences: Dispatch<Preferences[]>;
+  editState: undefined | string;
+  setEditState: Dispatch<undefined | string>;
 }) {
-  const updatePreferences = (currentPrefs: string[], pref: string) => {
-    var index = currentPrefs.indexOf(pref);
-    if (index >= 0) currentPrefs.splice(index, 1);
-    else currentPrefs.push(pref);
-    return currentPrefs;
-  };
+  const isRepresentative = type == "Representative";
+  const defaultPreference = new Preferences(undefined, "", [], "", type);
+
+  const [checkmarks, setCheckMarks] = useState<boolean[]>([
+    false,
+    false,
+    false,
+    false,
+  ]);
+  const [preference, setPreference] = useState(preferences[pos]);
+
+  const setPreferenceMutation = api.exhibitor.setFoodPreferences.useMutation();
+  const deletePreferenceMutation =
+    api.exhibitor.deleteFoodPreferences.useMutation();
+
+  function convertCheckMarks(checkmarks: boolean[]): Options[] {
+    const options: Options[] = ["Meat", "Vegan", "LactoseFree", "GlutenFree"];
+    return checkmarks.map((_, i) => options[i]).filter((_, i) => checkmarks[i]);
+  }
+
+  function handleSubmission(e: any) {
+    e.preventDefault();
+    setPreferenceMutation.mutate({
+      id: preference.id,
+      name: preference.name,
+      value: preference.value,
+      comment: preference.comment,
+      type: preference.type,
+    });
+    setEditState(undefined);
+  }
+
+  function deletePreferenceInDatabase() {
+    deletePreferenceMutation.mutate({
+      id: preferences[pos].id,
+      locale: t.locale,
+    });
+    setEditState(undefined);
+  }
+
+  useEffect(() => {
+    const pref = preferences[pos < preferences.length ? pos : 0];
+    setPreference(pref);
+    setCheckMarks([
+      pref.value.includes("Meat"),
```

Comment:
![image](https://github.com/datasektionen/ddagen/assets/32957299/f0a1afd6-12b9-4901-ab86-1028a06b8838)
I get this error when deleting preferences. Seems to happen all the time.